pkg/extension/sumologicextension/extension.go

```diff
@@ -57,20 +60,30 @@ const (
 )
 
 func newSumologicExtension(conf *Config, logger *zap.Logger) (*SumologicExtension, error) {
+	collectorName := ""
+	uuid := uuid.New()
+	hostname, err := os.Hostname()
+	if err != nil {
+		return nil, err
+	}
 	if conf.CollectorName == "" {
-		return nil, errors.New("collector name is unset")
+		conf.CollectorName = fmt.Sprintf("%s-%s", hostname, uuid.String())
```

Comment:
If I restart collector, is a new one created, or the previous name is being used?

Reply:
@pmatyjasek-sumo is working on this. The idea is that when you don't specify the collector name (i.e. `""`) you'll get one specific name (based on hostname as far as I recall correctly) and yes it should be the same on each subsequent run with the same empty collector name.